### ISSUE TITLE
bun install --latest  is not installing the latest

### ISSUE THREAD

**Issue by dylang** [oven-sh/bun#5506]
### What version of Bun is running?

1.0.2+37edd5a6e389265738e89265bcbdf2999cb81a49

### What platform is your computer?

Darwin 22.6.0 x86_64 i386

### What steps can reproduce the bug?

`parse-json@5.2.0` has [a dependency on lines-and-columns `^1.1.6`](https://github.com/sindresorhus/parse-json/blob/b778e2eb368a0b935e0a999f6366e9edd95e4dc7/package.json#L38)

Install with no lock file or cache. 

```sh
bun pm cache rm && rm -rf bun.lockb yarn.lock node_modules && bun install --latest
```

### What is the expected behavior?

Bun installs the latest compatible version to `^1.1.6`, which is `lines-and-columns@1.2.4`.

### What do you see instead?

Bun installs `lines-and-columns@1.2.0`, which happens to be a broken version. 

This happens every time in our monorepo, but not standalone.

The package that leads to this is [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/package.json), but installing it in other monorepos doesn't cause this problem. There are no packages that depend on `lines-and-columns@1.2.0`, as you can see in the yarn.lock output.

### Additional information

From the yarn.lock, these are the only entries for both modules. 

This is AFTER the install runs with no lock file and no cache.

```
lines-and-columns@^1.1.6:
  version "1.2.0"
  resolved "https://artifactory.company.com/api/npm/virtual-npm/lines-and-columns/-/lines-and-columns-1.2.0.tgz"
  integrity sha512-.....
```
....
```
parse-json@^5.0.0, parse-json@^5.2.0:
  version "5.2.0"
  resolved "https://artifactory.company.com/api/npm/virtual-npm/parse-json/-/parse-json-5.2.0.tgz"
  integrity sha512-....
  dependencies:
    "@babel/code-frame" "^7.0.0"
    error-ex "^1.3.1"
    json-parse-even-better-errors "^2.3.0"
    lines-and-columns "^1.1.6"
```

yarn is using the same registry and creditials as bun.

```
yarn info lines-and-columns
yarn info v1.22.19
{
  name: 'lines-and-columns',
  description: 'Maps lines and columns to character offsets and back.',
  'dist-tags': {
    v11x: '1.1.11',
    v12x: '1.2.4',
    latest: '2.0.3'
  },
  versions: [
    '1.0.0',
    '1.0.1',
    '1.0.2',
    '1.1.0',
    '1.1.1',
    '1.1.2',
    '1.1.3',
    '1.1.4',
    '1.1.5',
    '1.1.6',
    '1.1.7',
    '1.1.8',
    '1.1.9',
    '1.1.10',
    '1.1.11',
    '1.2.0',  // <-- Why is this version being installed?
    '1.2.1',
    '1.2.2',
    '1.2.3',
    '1.2.4',  // <-- Instead of this version?
    '2.0.0',
    '2.0.1',
    '2.0.2',
    '2.0.3'
  ],
  maintainers: [
    {
      name: 'eventualbuddha',
      email: 'brian@donovans.cc'
    }
  ],
  author: {
    name: 'Brian Donovan',
    email: 'brian@donovans.cc'
  },
  time: {
    modified: '2022-06-19T12:31:51.246Z',
    '2.0.0': '2021-11-20T22:27:07.947Z',
    '2.0.1': '2021-11-20T22:42:19.703Z',
    '1.1.10': '2021-11-21T00:32:46.542Z',
    created: '2015-12-02T19:57:59.708Z',
    '1.1.11': '2021-11-21T03:30:36.129Z',
    '2.0.2': '2021-11-20T23:28:46.961Z',
    '2.0.3': '2021-11-23T03:15:17.010Z',
    '1.1.2': '2016-06-24T03:36:31.344Z',
    '1.2.1': '2021-11-20T23:17:19.625Z',
    '1.1.3': '2016-06-24T03:52:38.515Z',
    '1.2.2': '2021-11-21T00:28:55.542Z',
    '1.1.4': '2016-06-26T16:57:46.382Z',
    '1.2.3': '2021-11-21T00:30:55.831Z',
    '1.1.5': '2016-06-26T21:54:01.493Z',
    '1.2.4': '2021-11-21T03:31:07.871Z',  // <-- Instead of this version?
    '1.1.6': '2016-12-20T17:32:14.921Z',
    '1.1.7': '2021-11-20T22:15:14.474Z',
    '1.1.8': '2021-11-20T23:09:10.780Z',
    '1.1.9': '2021-11-20T23:15:03.350Z',
    '1.0.0': '2015-12-02T19:57:59.708Z',
    '1.0.1': '2015-12-02T20:00:17.654Z',
    '1.1.0': '2016-03-09T01:49:47.038Z',
    '1.0.2': '2015-12-05T19:32:05.157Z',
    '1.1.1': '2016-03-09T01:56:07.519Z',
    '1.2.0': '2021-11-20T22:23:35.670Z'.  // <-- Why is this version being installed?
  },
  repository: {
````

**Comment by bitfrost** [comment]
I am seeing the same thing with this same exact package and version.  However only when using a internal private registry (nexus instead of artifactory in the example above). When I switch to public npm registry I correctly get 1.2.4

**Comment by manzt** [comment]
I might be running into something similar with `bun create`. I have a create CLI, which bun fetched the latest version at the time from npm (0.4.0):

```
bun create anywidget@latest
```

but then I released a new version (0.4.1) and bun uses the cached version (0.4.0). I tried pruning the cache and even deleted bun and it still will fetch `0.4.0`.

**Comment by deemkeen** [comment]
is there any workaround for this? package override doesn't seem to work either..
